cost-analyzer/values.yaml

```diff
@@ -1165,9 +1167,32 @@ prometheus:
     ##
     terminationGracePeriodSeconds: 300
 
-    ## Prometheus data retention period (default if not specified is 15 days)
+    ## Prometheus data retention period (default if not specified is 97 hours)
     ##
-    retention: 15d  # 50h. This must be greater than or equal to etlHourlyStoreDurationHours
+    ## Kubecost builds up its own persistent store of metric data on the
+    ## filesystem (usually a PV) and, when using ETL Backup and/or Federated
+    ## ETL, in more durable object storage like S3 or GCS. Kubecost's data
+    ## retention is _not_ tied to the configured Prometheus retention.
+    ##
+    ## For data durability, we recommend using ETL Backup instead of relying on
+    ## Prometheus retention.
+    ##
+    ## Lower retention values will affect Prometheus by reducing resource
+    ## consumption and increasing stability. It _must not_ be set below or equal
+    ## to kubecostModel.etlHourlyStoreDurationHours, otherwise empty data sets
+    ## may overwrite good data sets. For now, it must also be >= 49h for Daily
```

Comment:
> it must also be >= 49h for Daily ETL stability.

Do you think it's worth making this a validation check so the chart cannot be deployed if it violates this rule?

Reply:
I like the idea. We can also do this in go code, but I think a helm chart check would be both a more logical place for this, AND would take a lot less code to implement

Reply:
I think this might be a pretty gnarly check, because retention can be set with Prometheus' string window syntax (e.g. `3h`, `3d`, `3m` - https://prometheus.io/docs/prometheus/latest/querying/basics/#time-durations). We'd have to either:
1. Implement gnarly logic supporting common duration syntax
2. Force this duration to be specified in hours

2 is probably the most palatable, but I want to raise a flag that it won't be _super_ straightforward in Helm. I'll get to work on 2 and we can see what it looks like before making a decision.

Reply:
Added such a check in the latest commit https://github.com/kubecost/cost-analyzer-helm-chart/pull/3167/commits/fe0b417cf9571023bff2ef79452a1741279897eb -- see below for test notes.

Reply:
Nice work, Michael. I approve.

Reply:
Nice work on the Prometheus metric retention check!